### PR TITLE
Objects and Object Constructors: Add explanation to distinguish b/w Constructor.prototype vs obj.__proto__

### PR DESCRIPTION
**Pull request by anupamvamsi**
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

The primary aim of this PR was to update the lesson's **Recommended Method for Prototypal Inheritance** content from using `Object.create()` to the `Object.setPrototypeOf()` method.

However I realized that the lesson also would need to better explain the `prototype` function property vs. the `__proto__` /  `[[Prototype]]` property and prototypal inheritance in this lesson, as personally to me as a learner, it was confusing and I had to refer to quite a few resources to understand it.

This is the first PR in an upcoming set of PRs to make the review process easier as suggested in the Discord forum post.

**2. This PR:**

* Explains the slight differences between the `Object.prototype` and the `obj.__proto__` concepts
* It also identifies that the `__proto__` way of referring to the `.prototype` property is now done using `Object.getPrototypeOf()` method.
* Provides examples to illustrate that `obj.__proto__` (or) `Object.getPrototypeOf(obj)` refers to `Object.prototype`.
* Recognizes that `__proto__` is also referred to as `[[Prototype]]` in the specification for clarity.
* Puts forth the idea that `__proto__` can have a `null` value and will be explained further in the lesson, and the explanation will be brought in through further edits (and PRs).

**3. Additional Information:**

* Suggestions-Bugs Forum discussion on Discord: https://discord.com/channels/505093832157691914/1010249197544538112
* Discussion on mdn/content: https://github.com/mdn/content/pull/18880 
* Next PR: #24563.

### CODE REVIEW

**Comment by wise-king-sullyman** [comment]
Hey @anupamvamsi! I wanted to pop in real quick, say thanks for putting up these PR's, and let you know that this _is_ under review.

As I'm sure you're super aware after doing the research to put this together, the real "in the weeds" specifics of `prototype` in JS are (in my opinion) fairly convoluted and hard to clearly discuss even between people that are experienced in the language, not to mention trying to explain it to people that are still in the early stages of learning and may not have any OOP experience to use as a reference when discussing the behavior.

For that reason, the review process on this has been pretty slow, but I hope to have some comments back to you within the next few days. Broadly I like the ideas put forward, but I might have some suggestions to reword some things for the sake of clarity 🙂

**Comment by anupamvamsi** [comment]
Hi @wise-king-sullyman, thanks for letting me know! I did have another part to this PR, which I have not yet pushed. Would it be fine if I pushed it for review as well?

**Comment by wise-king-sullyman** [comment]
@anupamvamsi yes that should be fine

File: javascript/organizing_your_javascript_code/objects_and_object_constructors.md [hunk 1]
```diff
@@ -151,6 +151,45 @@ console.log(theHobbit.info());
 
 Before we go much further, there's something important you need to understand about JavaScript objects. All objects in JavaScript have a `prototype`. Stated simply, the prototype is another object that the original object _inherits_ from, which is to say, the original object has access to all of its prototype's methods and properties.
 
+To elaborate, there are two kinds of `prototype`, which are intertwined with each other. The _first_ kind is the one that exists on a **function**, which is referred to as `Player.prototype` in the case of the `Player` function in the previous example.
```
Comment (wise-king-sullyman) [inline javascript/organizing_your_javascript_code/objects_and_object_constructors.md:154]: I think I get what you're going for in saying there are two "kinds" of prototype, but I feel like the way things are phrased isn't *quite* accurate. There's only one "kind" of prototype, the actual prototype of something which in the spec is referred to as `[[prototype]]`, but there's also the `.prototype` _property_ which behaves in a similar kind of way but has distinctly different rules as to what that property is.

I suppose I'm just not sure about referring to this as two "kinds" of prototype, since really one is the prototype, and the other is the value of the `.prototype` property, if that makes sense.

Do you think there might be a way you can reword this to make that differentiation more clear?

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
Apologies for the delay! I have one question/suggestion that I'd like to know your thoughts on regarding how `prototype` is being discussed here.

**Comment by anupamvamsi** [comment]
It's been a while since I looked at this, honestly, so I'm looking at all this with a fresher, newer perspective? When I read it all again, overall, I do feel there's some things that can be improved - in addition to the points you have mentioned.

I'll try making it more clear. Hopefully by keeping it simple.

One thing I could use your advice on though -- #24563 (the next PR) adds on to the commits I've made in this PR. And #24782 (the latest one) adds on to the commits present in #24563. 

1. So, if I make changes, they would affect all the three PRs. 

2. Would it be okay if I make a new PR with all the changes that I am thinking of currently? This new PR would include the commits present in #24782 (the latest one). There are three PRs currently as I was told that separate PRs with small changes would be good and easy to review, but to me, the way I've submitted them seems to have defeated that purpose. My apologies if these PRs have been submitted in a messy way, if there's something I can do about that, please do let me know!

**Comment by wise-king-sullyman** [comment]
@anupamvamsi yeah if you want to just make a new PR with all the changes you're thinking of and close your open PRs that would be fine!

And yeah generally when you want to make multiple PRs for something like this it's often best to make each one after the previous one has been merged so that you don't run into these kinds of issues. Another option would be to just rebase those other branches after your updates here are merged, but that adds an extra level of Git complication that many people aren't super comfortable with.

Absolutely no worries btw, I can recognize that you're just trying to do things in the best way 🙂

**Comment by anupamvamsi** [comment]
#24941 closes this.
